watertap/examples/flowsheets/case_studies/electroNP/BSM2_electroNP_no_bioP.py

```diff
@@ -0,0 +1,666 @@
+    Product,
+    Mixer,
+    PressureChanger,
+)
+from idaes.models.unit_models.separator import SplittingType
+from idaes.core.solvers import get_solver
+from idaes.core.util.model_statistics import degrees_of_freedom
+import idaes.logger as idaeslog
+import idaes.core.util.scaling as iscale
+from idaes.core.util.tables import (
+    create_stream_table_dataframe,
+    stream_table_dataframe_to_string,
+)
+from watertap.unit_models.cstr_injection import CSTR_Injection
+from watertap.unit_models.clarifier import Clarifier
+from watertap.property_models.anaerobic_digestion.modified_adm1_properties import (
+    ModifiedADM1ParameterBlock,
+)
+from watertap.property_models.anaerobic_digestion.adm1_properties_vapor import (
+    ADM1_vaporParameterBlock,
+)
+from watertap.property_models.anaerobic_digestion.modified_adm1_reactions import (
+    ModifiedADM1ReactionParameterBlock,
+)
+from watertap.property_models.activated_sludge.modified_asm2d_properties import (
+    ModifiedASM2dParameterBlock,
+)
+from watertap.property_models.activated_sludge.modified_asm2d_reactions import (
+    ModifiedASM2dReactionParameterBlock,
+)
+from watertap.unit_models.translators.translator_adm1_asm2d import (
+    Translator_ADM1_ASM2D,
+)
+from idaes.models.unit_models.mixer import MomentumMixingType
+from watertap.unit_models.translators.translator_asm2d_adm1 import (
+    Translator_ASM2d_ADM1,
+)
+from watertap.unit_models.anaerobic_digester import AD
+from watertap.unit_models.dewatering import (
+    DewateringUnit,
+    ActivatedSludgeModelType as dewater_type,
+)
+from watertap.unit_models.thickener import (
+    Thickener,
+    ActivatedSludgeModelType as thickener_type,
+)
+from watertap.core.util.initialization import check_solve
+from watertap.unit_models.electroNP_ZO import ElectroNPZO
+
+# Set up logger
+_log = idaeslog.getLogger(__name__)
+
+
+def main(has_electroNP=False):
+    m = build_flowsheet(has_electroNP=has_electroNP)
+    set_operating_conditions(m)
+    for mx in m.fs.mixers:
+        mx.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 3].deactivate()
+    print(f"DOF before initialization: {degrees_of_freedom(m)}")
+
+    initialize_system(m)
+    for mx in m.fs.mixers:
+        mx.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 2].deactivate()
+    m.fs.MX3.pressure_equality_constraints[0.0, 3].deactivate()
+    print(f"DOF after initialization: {degrees_of_freedom(m)}")
+
+    results = solve(m)
+
+    pyo.assert_optimal_termination(results)
+    check_solve(
+        results,
+        checkpoint="re-solve with controls in place",
+        logger=_log,
+        fail_flag=True,
+    )
+
+    return m, results
+
+
+def build_flowsheet(has_electroNP=False):
+    m = pyo.ConcreteModel()
+
+    m.fs = FlowsheetBlock(dynamic=False)
+    m.fs.has_electroNP = has_electroNP
```

Comment:
Is it necessary to have this? I assume if someone is running this flowsheet, they'd want ElectroN-P to be included. If they didn't want ElectroN-P, they could just run the Extended BSM2 flowsheet. Was this just leftover from when you were diagnosing issues with the flowsheet?